### PR TITLE
Start refactoring bits of deabstraction

### PR DESCRIPTION
**Pull request by lattner**
- introduce a new pass to clean up instructions instead of having
   all the transformations do it.  This needs to be generalized, but is
   a start.  This should improve compile time when strict deabstraction
   is on.
 - Refactor a bunch of code into a "createConstTensor" helper in
   TFUtilities.  It would be nice for the rest of the compiler that
   produces "Const" nodes to start using this.

The patch to SILType.h has been sent upstream in PR18003 so it has no
SWIFT_ENABLE_TENSORFLOW marker.

### CODE REVIEW

**Comment by lattner** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFUtilities.cpp [hunk 1]
```diff
@@ -1899,6 +1921,57 @@ SILLocation tf::getUserSourceLocation(SILInstruction *inst) {
...
+  auto shapeSuffix =
+  SILTensorOpInfo::getOperandClassSuffix(SILTensorOpInfo::OperandClass::Shape);
+  attributes.push_back({
+    context.getIdentifier(std::string("value") + shapeSuffix), shape
+  });
+
+  // All graph_op's get a device.
+  attributes.push_back({
+    context.getIdentifier(DEVICE_ATTR),
+    SymbolicValue::getString(getDeviceString(deviceConfig.primaryDeviceType),
```
Comment (mhong) [inline lib/SILOptimizer/Mandatory/TFUtilities.cpp]: use ALL_DEVICES here?

**Review by mhong — APPROVED** [review]

**Review by rxwei — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/TFUtilities.cpp (hunk 1 above)
Comment (lattner) [reply lib/SILOptimizer/Mandatory/TFUtilities.cpp]: This is intended to be usable by multiple clients.  Hard coding ALL_DEVICES doesn't seem general... wouldn't that prevent it from being useful to lowerGraph and other things that need it?

Comment (lattner) [reply lib/SILOptimizer/Mandatory/TFUtilities.cpp]: We chatted, I know what to do :-)

**Comment by lattner** [comment]
@swift-ci please test and merge tensorflow

**Comment by lattner** [comment]
@swift-ci please test tensorflow and merge

**Comment by lattner** [comment]
@swift-ci please test tensorflow

**Comment by lattner** [comment]
@swift-ci please test tensorflow

**Pull request merged**